### PR TITLE
Disable sending assets with zero balances

### PR DESCRIPTION
**Pull request by keshan3262**

### CODE REVIEW

File: src/components/header-card-action-buttons/header-card-action-buttons.styles.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { createUseStyles } from '../../styles/create-use-styles';
+
+export const useHeaderCardActionButtonsStyles = createUseStyles(() => ({
+  buttonContainer: {
```
Comment (unixvb) [inline src/components/header-card-action-buttons/header-card-action-buttons.styles.ts:4]: I think you should not use this for the `RECEIVE` and `BUY ` buttons.  
This does nothing to them.

File: src/components/header-card-action-buttons/header-card-action-buttons.tsx [hunk 2]
```diff
@@ -1,33 +1,58 @@
...
+          title="RECEIVE"
+          iconName={IconNameEnum.ArrowDown}
+          onPress={() => navigate(ModalsEnum.Receive, { asset })}
+        />
+      </View>
       <Divider size={formatSize(8)} />
-      <ButtonMedium title="SEND" iconName={IconNameEnum.ArrowUp} onPress={() => navigate(ModalsEnum.Send, { asset })} />
+      <View
+        onTouchStart={() => showErrorToast(`Can't send ${asset.symbol}`, errorMessage)}
+        style={styles.buttonContainer}>
```
Comment (unixvb) [inline src/components/header-card-action-buttons/header-card-action-buttons.tsx:44]: wrong handlers order

File: src/modals/send-modal/send-modal.tsx [hunk 3]
```diff
@@ -39,14 +39,15 @@ export const SendModal: FC = () => {
...
-    ],
+    () =>
+      [
+        {
+          ...emptyToken,
+          ...TEZ_TOKEN_METADATA,
+          balance: selectedAccount.tezosBalance.data
+        },
+        ...tokensList
+      ].filter(token => Number(token.balance) > 0),
```
Comment (unixvb) [inline src/modals/send-modal/send-modal.tsx:50]: Please, reuse logic `useFilteredTokenList` hook logic from `TokenList` component.  
Let it filter in the same way.

File: src/screens/tezos-token-screen/tezos-token-screen.tsx [hunk 4]
```diff
@@ -33,7 +33,7 @@ export const TezosTokenScreen = () => {
 
         <PublicKeyHashText publicKeyHash={selectedAccount.publicKeyHash} marginBottom={formatSize(16)} />
 
-        <HeaderCardActionButtons asset={TEZ_TOKEN_METADATA} />
+        <HeaderCardActionButtons asset={TEZ_TOKEN_METADATA} balance={tezosBalance} />
```
Comment (unixvb) [inline src/screens/tezos-token-screen/tezos-token-screen.tsx:36]: What do you think about creating `useTezosTokenSelector` that will return `TokenInterface`?
(the same what we are doing in `SendModal` right now)
```
{
          ...emptyToken,
          ...TEZ_TOKEN_METADATA,
          balance: selectedAccount.tezosBalance.data
        }
```
And you will be able to pass a single prop instead.

File: src/components/header-card-action-buttons/header-card-action-buttons.tsx [hunk 5]
```diff
@@ -1,21 +1,40 @@
...
+import { isDefined } from '../../utils/is-defined';
 import { ButtonMedium } from '../button/button-medium/button-medium';
 import { ButtonsContainer } from '../button/buttons-container/buttons-container';
 import { Divider } from '../divider/divider';
 import { IconNameEnum } from '../icon/icon-name.enum';
+import { useHeaderCardActionButtonsStyles } from './header-card-action-buttons.styles';
 
 interface Props {
-  asset: AssetMetadataInterface;
+  asset: TokenInterface;
```
Comment (unixvb) [inline src/components/header-card-action-buttons/header-card-action-buttons.tsx:19]: `asset: TokenInterface;` worong naming

File: src/components/header-card-action-buttons/header-card-action-buttons.tsx [hunk 6]
```diff
@@ -25,7 +44,14 @@ export const HeaderCardActionButtons: FC<Props> = ({ asset }) => {
         onPress={() => navigate(ModalsEnum.Receive, { asset })}
       />
       <Divider size={formatSize(8)} />
-      <ButtonMedium title="SEND" iconName={IconNameEnum.ArrowUp} onPress={() => navigate(ModalsEnum.Send, { asset })} />
+      <View style={styles.buttonContainer} onTouchStart={onTouchStart}>
```
Comment (unixvb) [inline src/components/header-card-action-buttons/header-card-action-buttons.tsx:47]: replace `onTouchStart` with beautiful short return anonymous function
```
onTouchStart={() => disableSendAsset && showErrorToast(`Can't send ${asset.symbol}`, errorMessage)}
```

File: src/store/wallet/wallet-selectors.ts [hunk 7]
```diff
@@ -55,6 +56,16 @@ export const useVisibleTokensListSelector = () => {
 
 export const useTezosBalanceSelector = () => useSelectedAccountSelector().tezosBalance.data;
 
+export const useTezosAssetSelector = () => {
```
Comment (unixvb) [inline src/store/wallet/wallet-selectors.ts:59]: please, set return type `TokenInterface`

File: src/modals/send-modal/send-modal.tsx [hunk 8]
```diff
@@ -37,26 +38,27 @@ export const SendModal: FC = () => {
...
+  const { filteredTokensList } = useFilteredTokenList(tokensList, true);
 
-  const tokensListWithTez = useMemo<TokenInterface[]>(
+  const filteredTokensListWithTez = useMemo<TokenInterface[]>(
     () => [
       {
         ...emptyToken,
         ...TEZ_TOKEN_METADATA,
         balance: selectedAccount.tezosBalance.data
       },
```
Comment (unixvb) [inline src/modals/send-modal/send-modal.tsx:49]: replace with `useTezosAssetSelector`

File: src/components/header-card-action-buttons/header-card-action-buttons.tsx (hunk 6 above)
Comment (keshan3262) [reply src/components/header-card-action-buttons/header-card-action-buttons.tsx:47]: Sorry, it doesn't work for mysterious reasons, you can check it yourself

File: src/components/header-card-action-buttons/header-card-action-buttons.tsx (hunk 5 above)
Comment (keshan3262) [reply src/components/header-card-action-buttons/header-card-action-buttons.tsx:19]: But which is correct?

Comment (unixvb) [reply src/components/header-card-action-buttons/header-card-action-buttons.tsx:19]: token: TokenInterface;

File: src/components/header-card-action-buttons/header-card-action-buttons.tsx (hunk 6 above)
Comment (unixvb) [reply src/components/header-card-action-buttons/header-card-action-buttons.tsx:47]: try to wrap it with void
`onTouchStart={() => void(disableSendAsset && showErrorToast(`Can't send ${asset.symbol}`, errorMessage))}`. 
 If this woult not work, let it be like this

File: src/store/wallet/wallet-selectors.ts (hunk 7 above)
Comment (unixvb) [reply src/store/wallet/wallet-selectors.ts:59]: and I think it should be `useTezosTokenSeletor` also

**Review by unixvb — APPROVED** [review]

**Review by andrewkryshtal — APPROVED** [review]

**Pull request merged**